tests/iree-pytorch/matmul/run.c

```diff
@@ -0,0 +1,143 @@
+#include <stddef.h>
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "iree/modules/hal/module.h"
+#include "iree/vm/api.h"
+#include "iree/vm/bytecode/module.h"
+
+// A function to create the HAL device from the different backend targets.
+// The HAL device is returned based on the implementation, and it must be
+// released by the caller.
+extern iree_status_t create_sample_device(iree_allocator_t host_allocator,
+    iree_hal_device_t **out_device);
+
+// A function to load the vm bytecode module from the different backend targets.
+// The bytecode module is generated for the specific backend and platform.
+extern const iree_const_byte_span_t load_bytecode_module_data();
+
+iree_status_t Run(void) {
+    iree_vm_instance_t* instance = NULL;
+    IREE_RETURN_IF_ERROR(iree_vm_instance_create(
+        IREE_VM_TYPE_CAPACITY_DEFAULT, iree_allocator_system(), &instance));
+    IREE_RETURN_IF_ERROR(iree_hal_module_register_all_types(instance));
+  
+    iree_hal_device_t* device = NULL;
+    IREE_RETURN_IF_ERROR(create_sample_device(iree_allocator_system(), &device),
+                         "create device");
+    iree_vm_module_t* hal_module = NULL;
+    IREE_RETURN_IF_ERROR(iree_hal_module_create(
+        instance, /*device_count=*/1, &device, IREE_HAL_MODULE_FLAG_SYNCHRONOUS,
+        iree_hal_module_debug_sink_stdio(stderr), iree_allocator_system(),
+        &hal_module));
+  
+    // Load bytecode module from the embedded data.
+    const iree_const_byte_span_t module_data = load_bytecode_module_data();
+  
+    iree_vm_module_t* bytecode_module = NULL;
+    IREE_RETURN_IF_ERROR(iree_vm_bytecode_module_create(
+        instance, module_data, iree_allocator_null(), iree_allocator_system(),
+        &bytecode_module));
+  
+    // Allocate a context that will hold the module state across invocations.
+    iree_vm_context_t* context = NULL;
+    iree_vm_module_t* modules[] = {hal_module, bytecode_module};
+    IREE_RETURN_IF_ERROR(iree_vm_context_create_with_modules(
+        instance, IREE_VM_CONTEXT_FLAG_NONE, IREE_ARRAYSIZE(modules), &modules[0],
+        iree_allocator_system(), &context));
+    iree_vm_module_release(hal_module);
+    iree_vm_module_release(bytecode_module);
+  
+    // Lookup the entry point function.
+    // Note that we use the synchronous variant which operates on pure type/shape
+    // erased buffers.
+    const char kMainFunctionName[] = "module.main";
```

Comment:
Any specific reason not to use "const char*"?

Reply:
https://github.com/iree-org/iree-bare-metal-arm/blob/9837bbaa9dd30f5569d544883e4b0492aa53bbda/samples/simple_embedding/simple_embedding_int.c#L68